clang/lib/Sema/SemaDecl.cpp

```diff
@@ -13435,16 +13435,18 @@ void Sema::checkNonTrivialCUnion(QualType QT, SourceLocation Loc,
 void Sema::AddInitializerToDecl(Decl *RealDecl, Expr *Init, bool DirectInit) {
   // If there is no declaration, there was an error parsing it.  Just ignore
   // the initializer.
-  if (!RealDecl || RealDecl->isInvalidDecl()) {
+  if (!RealDecl) {
     CorrectDelayedTyposInExpr(Init, dyn_cast_or_null<VarDecl>(RealDecl));
     return;
   }
 
   if (CXXMethodDecl *Method = dyn_cast<CXXMethodDecl>(RealDecl)) {
```

Comment:
Minor nit: We can use separate init-statement since we adopted C++17; You call.
```suggestion
  if (auto *Method = dyn_cast<CXXMethodDecl>(RealDecl); !Method->isInvalidDecl()) {
```

Reply:
Note, I believe in this formulation it needs to be:

```
if (auto *Method = dyn_cast<CXXMethodDecl>(RealDecl); Method && !Method->isInvalidDecl())
```

Reply:
Actually, that's not quite right, either. There is a `return;` that we want to take for any method, and a diagnostic for the `!Method->isInvalidDecl()` case. We therefore do need two nested `if`s.